### ISSUE TITLE
Issues with prisma client in 1.25

### ISSUE THREAD

**Issue by n2ctech** [prisma/prisma#3919]
**Describe the bug**
All method chaining with prisma client now produce a bug

**To Reproduce**
Use method chaining inside a custom resolver. Get an obscure error (see screenshots)

**Expected behavior**
Chaining like ctx.db.student({ id }).school().name() used to work and return the scalar value.

**Screenshots**
Working code before 1.25 : 
<img width="312" alt="screenshot 2019-01-23 at 18 59 26" src="https://user-images.githubusercontent.com/21331958/51664269-61b25a00-1fb9-11e9-94b0-b040f1f65561.png">
After upgrading prisma, have to revert to using bindings for it to work now, e.g. in the first resolver : 
<img width="390" alt="screenshot 2019-01-24 at 09 34 03" src="https://user-images.githubusercontent.com/21331958/51665089-3df01380-1fbb-11e9-860e-ed195db6c008.png">

New errors from the front everywhere when calling a resolver using method chaining in the back : 
<img width="999" alt="screenshot 2019-01-23 at 18 57 46" src="https://user-images.githubusercontent.com/21331958/51664346-89a1bd80-1fb9-11e9-94c1-b38d52b205d2.png">

**Versions (please complete the following information):**
 - Connector: `MySQL`
 - Prisma Server:  `1.25.0` (fargate)
 - `prisma` CLI: [e.g. `prisma/1.25.1 (darwin-x64) node-v10.4.0`]
 - OS:  `OS X High Sierra`
 - other dependencies: `prisma-client`, `prisma-binding`

Is this a known issue ? Do I have to convert all uses of the prisma client to good old bindings (a lot of places in my code) ?

**Comment by pantharshit00** [comment]
Hi @n2ctech 

I am unable to reproduce this with the following code:
```js
const { prisma } = require('./generated/prisma-client')

;(async () => {
  const data = await prisma.user({ id: 'cjrahb9ul000i0808zqcpngsg' }).post().like()
  console.log(data);
})()

```

datamodel:
```graphql
type User {
  id: ID! @unique
  name: String!
  post: Post!
}

type Post {
  id: ID! @unique
  content: String
  user: User!
  like: Like
}

type Like {
  id: ID! @unique
  liked: Boolean
  post: Post
}
```

![image](https://user-images.githubusercontent.com/22195362/51673155-4feda880-1ff3-11e9-961a-dfcc49d4e9ad.png)
This works fine.

I am using prisma-client-lib version 1.25.2 so please upgrade to it.

Please provide a reproduction repo if you are still facing this.

**Comment by n2ctech** [comment]
@pantharshit00  thanks, can you try finishing the chaining on a scalar (e.g. `prisma.user({ id: 'cjrahb9ul000i0808zqcpngsg' }).post().content()`)

**Comment by matthewsecrist** [comment]
I'm able to reproduce this. I'm working through https://www.howtographql.com/graphql-js/8-filtering-pagination-and-sorting/ and ran into the error message at the Count part. Was able to narrow it down to Prisma, and specifically with 1.25.

Clone the repo https://github.com/howtographql/graphql-js, `yarn install`

I changed the /prisma/prisma.yml to be:

```yml
endpoint: http://localhost:4466
datamodel: datamodel.prisma

generate:
  - generator: javascript-client
    output: ../src/generated/prisma-client

hooks:
  post-deploy:
    - prisma generate
```

and then `prisma deploy` and `yarn start`. Everything works fine. Update the prisma-client-lib from 1.23 to 1.25.3 and it fails.

```json
{
  "data": null,
  "errors": [
    {
      "message": "Field 'count' of type 'Int' must not have a sub selection. (line 4, column 7):\n      count {\n      ^",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "feed"
      ]
    }
  ]
}
```

**Comment by ahrbil** [comment]
> I'm able to reproduce something like this. I'm working through https://www.howtographql.com/graphql-js/8-filtering-pagination-and-sorting/ and ran into the error message at the Count part. Was able to narrow it down to Prisma, and specifically with 1.25.
> 
> Clone the repo https://github.com/howtographql/graphql-js, `yarn install`
> 
> I changed the /prisma/prisma.yml to be:
> 
> ```yaml
> endpoint: http://localhost:4466
> datamodel: datamodel.prisma
> 
> generate:
>   - generator: javascript-client
>     output: ../src/generated/prisma-client
> 
> hooks:
>   post-deploy:
>     - prisma generate
> ```
> 
> and then `prisma deploy` and `yarn start`. Everything works fine. Update the prisma-client-lib from 1.23 to 1.25.3 and it fails.
> 
> ```json
> {
>   "data": null,
>   "errors": [
>     {
>       "message": "Field 'count' of type 'Int' must not have a sub selection. (line 4, column 7):\n      count {\n      ^",
>       "locations": [
>         {
>           "line": 2,
>           "column": 3
>         }
>       ],
>       "path": [
>         "feed"
>       ]
>     }
>   ]
> }
> ```
> 
> The issue seems to be..
> 
> ```js
> const count = await context.prisma
>     .linksConnection()
>     .aggregate()
>     .count()
> ```
> 
> as replacing it with `const count = 5` has no error.

same for me after upgrading to 1.25.3
if using fragments it works
```
 const {
      aggregate: { count },
    } = await context.prisma
      .answersConnection({
        where: { answeredTo: { id: parent.id } },
      })
      .$fragment(fragment);
```

**Comment by nickluger** [comment]
Yup, can confirm this issue is a regression `1.25`. 

Ending the promise chain with requesting a scalar field like `db.user({ id: "123"}).age()`, results in 
```
field 'age' of type 'Int' must not have a sub selection.
```

Downgrade to `1.24` for a temporary workaroaund.

**Comment by divyenduz** [comment]
This is fixed as a patch in the latest stable (`1.25.4`). Please install it via `yarn add prisma-client-lib`. Please let me know if that works for you. Thanks.
